Controllers/StockController.cs

```diff
@@ -0,0 +1,282 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text.Json;
+using InvestingOak.Converters;
+using InvestingOak.Models.Finnhub;
+using Microsoft.AspNetCore.Mvc;
+
+namespace InvestingOak.Controllers
+{
+    [Route("api/stocks")]
+    [ApiController]
+    [Produces("application/json")]
+    [ResponseCache(VaryByHeader = "User-Agent", Duration = 30)]
+    public class StockController : ControllerBase
+    {
+        private const string AlphaVantageKey = "HNAFS0NKXJHEG5YI";
+        private const string AlphaVantageBaseUrl = "https://www.alphavantage.co/query";
+        private const string FinnhubKey = "btt5puv48v6q0kg1m610";
+        private const string FinnhubBaseUrl = "https://finnhub.io/api/v1";
+        private readonly HttpClient alphaVantageClient;
+        private readonly HttpClient finnhubClient;
+        private readonly JsonSerializerOptions serializerOptions;
+
+        public StockController()
+        {
+            // Configure HttpClients
+            alphaVantageClient = new HttpClient();
+            finnhubClient = new HttpClient();
+            finnhubClient.DefaultRequestHeaders.Add("X-Finnhub-Token", FinnhubKey);
+
+            // Configure JsonSerializer
+            serializerOptions = new JsonSerializerOptions
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                Converters =
+                {
+                    new Int32Converter(),
+                    new LongConverter(),
+                    new DecimalConverter(),
+                    new DateTimeOffsetConverter()
+                }
+            };
+        }
+
+        [HttpGet]
+        public ActionResult Symbols(string exchange)
+        {
+            try
+            {
+                var parameters = $"https://finnhub.io/api/v1/stock/symbol?exchange={exchange.ToUpperInvariant()}";
+
+                HttpResponseMessage response = finnhubClient.GetAsync(parameters).Result;
+                if (!response.IsSuccessStatusCode)
+                {
+                    throw new Exception($"Status code: {response.StatusCode}");
+                }
+
+                List<StockSymbol> stockSymbol =
+                    response.Content.ReadFromJsonAsync<List<StockSymbol>>(serializerOptions).Result;
+
+                return Ok(stockSymbol);
+            }
+            catch (Exception e)
```

Comment:
Codacy found an issue: [Catch a list of specific exception subtype or use exception filters instead.](https://app.codacy.com/gh/InvestingOak/InvestingOak/pullRequest?prid=6380270)